### PR TITLE
Discovery text tweaks.

### LINKED ISSUE #8: Discovery requirements
Do we want to add any text about how clients discover proxies. There are a few ways this works today, HTTP often uses WPAD that I finally found an IETF contribution for (!) https://tools.ietf.org/html/draft-ietf-wrec-wpad-01. Although that document didn't get adopted, it led me to find [RFC 3040](https://datatracker.ietf.org/doc/html/rfc3040) which has some nice descriptions of Web-oriented proxies.

### PR DESCRIPTION
**Pull request by LPardue**
This is based on PR #15 

fixes #8

### CODE REVIEW

File: draft-kuehlewind-quic-substrate.md [hunk 1]
```diff
@@ -70,14 +70,15 @@ be used for many other kinds of traffic, whenever the features provided by QUIC
...
-visible to one or both of the endpoints. If QUIC provides the basis for future tunneling
-and proxying solutions, it is expected that this relationship will change. At least one
-of the endpoints will be aware of the proxy and explicitly coordinate with it. This allows
-client hosts to make explicit decisions about the services they request from proxies
-(for example, simple forward or more advance performance-optimizing services),
-and to do so using a secure communication channel between themselves and the proxy.
+Existing proxies that are not based on QUIC are often transparent. That is, they
+do not require the cooperation of the ultimate connection endpoints, and are
+often not visible to one or both of the endpoints. QUIC security makes the
+transparent deployment model more difficult, therefore, it is expected that the
```
Comment (mirjak) [inline draft-kuehlewind-quic-substrate.md]: Not sure I like this phrasing. This point is true as well but my point was rather, that quic is a nice transport (with multiplexing support and build-in encryption) that is a very good candidate for a more generic explicit tunnel set up. I guess your argument flows better though. Maybe we can re-add my point somehow again?

File: draft-kuehlewind-quic-substrate.md [hunk 2]
```diff
@@ -277,21 +278,22 @@ Communication between the client and proxy is more likely to be realized as a
...
-server (CNAME/IPaddress). In both cases the proxy need to have a routable address
-and name.
+With QUIC, the use of proxying functions cannot be done transparently. Instead,
+proxies needs to be explicity discoverable. The simplest form of such discovery
+could include pre-configuration or via out-of-band signaling. The proxy could
+also be discovered through advertisement when a client is connected to a network
+(for example, the Dynamic Host Configuration Protocol). Alternatively, the
+client could obtain a white-listed proxy address when making first contact with
+the server (CNAME/IPaddress). In both cases the proxy needs to have a routable
+address and name.
```
Comment (mirjak) [inline draft-kuehlewind-quic-substrate.md:292]: This is good! Thanks!

**Review by mirjak — CHANGES_REQUESTED** [review]

**Comment by mirjak** [comment]
Also there seems to be merge conflicts. Can you please check @LPardue ? Thanks!

File: draft-kuehlewind-quic-substrate.md (hunk 1 above)
Comment (LPardue) [reply draft-kuehlewind-quic-substrate.md]: Noted. I split the commits on the PR and will address this on https://github.com/mirjak/draft-kuehlewind-quic-substrate/pull/15/files

**Comment by LPardue** [comment]
PTAL, should be more of a standalone change (git fu).

**Pull request merged**